src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/EligibilityAndSuitabilityService.kt

```diff
@@ -33,13 +38,24 @@ class EligibilityAndSuitabilityService(
   fun getCaseView(prisonNumber: String): EligibilityAndSuitabilityCaseView {
     val assessment = assessmentService.getCurrentAssessment(prisonNumber)
 
+    val eligibility = assessment.eligibilityProgress()
+    val eligibilityStatus = eligibility.toStatus()
+    val suitability = assessment.suitabilityProgress()
+    val suitabilityStatus = suitability.toStatus()
+
     return EligibilityAndSuitabilityCaseView(
       assessmentSummary = createAssessmentSummary(assessment),
       overallStatus = assessment.calculateAggregateEligibilityStatus(),
-      eligibility = assessment.eligibilityProgress(),
-      eligibilityStatus = assessment.eligibilityProgress().toStatus(),
-      suitability = assessment.suitabilityProgress(),
-      suitabilityStatus = assessment.suitabilityProgress().toStatus(),
+      eligibility = eligibility,
+      eligibilityStatus = eligibilityStatus,
+      suitability = suitability,
+      suitabilityStatus = suitabilityStatus,
+      failureType = when {
```

Comment:
Will this logic have to be done everywhere we construct one of these or can it be moved into the constructor?

Reply:
That's a really good question - The use of extension functions make this a bit misleading.

I tried to refactor this previously but the statuses/progress are currently dynamically generated based on having access to the policy info. 

I think maybe eligibility status checks should be persisted rather than dynamically generated